dijkstra-prim-visualization/src/components/Canvas/Canvas.jsx

```diff
@@ -1,115 +1,115 @@
+   * @param {MouseEvent} event - The click event.
+   */
   const canvasClickHandler = (event) => {
-    // Reset the firstNode state variable
-    setFirstNode(resetNodeData);
+    setFirstClickedNode(resetFirstClickedNode);
 
     // Calculate the coordinates of the clicked point relative to the canvas
-    const nodePointAbsoluteX = event.clientX;
-    const nodePointAbsoluteY = event.clientY;
-    const nodePointCanvasRelativeX =
-      nodePointAbsoluteX - canvasRef.current.getBoundingClientRect().left;
-    const nodePointCanvasRelativeY =
-      nodePointAbsoluteY - canvasRef.current.getBoundingClientRect().top;
-
-    // Create a new node point object
-    const newNodePoint = {
-      x: nodePointCanvasRelativeX,
-      y: nodePointCanvasRelativeY,
+    const nodeAbsoluteX = event.clientX;
+    const nodeAbsoluteY = event.clientY;
+    const nodeCanvasRelativeX =
+      nodeAbsoluteX - canvasRef.current.getBoundingClientRect().left;
+    const nodeCanvasRelativeY =
+      nodeAbsoluteY - canvasRef.current.getBoundingClientRect().top;
+
+    const newNode = {
+      id: `${nodeCanvasRelativeX}:${nodeCanvasRelativeY}`,
+      x: nodeCanvasRelativeX,
+      y: nodeCanvasRelativeY,
     };
 
     // Check if the new node position is valid (not overlapping with existing nodes)
-    if (!newNodePositionValid(newNodePoint, nodePoints, canvasRef)) {
+    if (!newNodePositionValid(newNode, nodes, canvasRef)) {
       return;
     }
 
-    // Add the new node point to the list of all nodes
-    setNodePoints((prevNodePoints) => [...prevNodePoints, newNodePoint]);
+    setNodes((prevNodes) => [...prevNodes, newNode]);
   };
 
-  // Handler function for when a node is clicked
-  const nodeClickHandler = (event, points) => {
+  /**
+   * Handler function for when a node is clicked.
+   * Adds a new edge between the first and second nodes if a second node is clicked.
+   * Resets the first node if the same node is clicked again.
+   * @param {MouseEvent} event - The click event.
+   * @param {Object} node - The clicked node object.
+   */
+  const nodeClickHandler = (event, node) => {
     event.stopPropagation();
 
-    // If no first node has been clicked yet, set the clicked node as the first node
-    if (firstNode.isClicked === false) {
-      setFirstNode({ isClicked: true, x: points.x, y: points.y });
-    }
-    // If the same node is clicked again, reset the first node
-    else if (firstNode.x === points.x && firstNode.y === points.y) {
-      console.log("first node clicked again, reset firstNode");
-      setFirstNode(resetNodeData);
-    }
-    // If a different node is clicked, create a new edge between the first and second nodes
-    else {
-      addEdge(firstNode, { x: points.x, y: points.y });
-      setFirstNode(resetNodeData);
-    }
-  };
+    const addEdge = (firstNode, secondNode) => {
+      const newEdge = {
+        id: `${firstNode.id}-${secondNode.id}`,
```

Comment:
Continuing the comment for node IDs, in this case, we can use the same format (so no changes are required), it will be then something like 3-5 which means an edge between the 4th and the 6th nodes.

Reply:
same here